### DISCUSSION TITLE
CHI_UNSPECIFIED while CHI_TETRAHEDRAL_CW and CHI_TETRAHEDRAL_CCW expected

### DISCUSSION THREAD

**Discussion by gokhantahil** [rdkit/rdkit#6291] [Q&A] [answered]
Hello, 

`SMILES = "C1=C[C@H]2[C@@H]3[C@H]([C@@H]1[C@H]4[C@@H]2[C@@H]5[C@H]4C6C=C[C@H]7C5[C@H]7C=C6)C=C[C@@H]8[C@H]3[C@@H]9[C@H]8C1C=C[C@H]2C9[C@@H]2C=C1"`
For the previous SMILES, with RDKit version 2021.03.5, the following code returns CHI_UNSPECIFIED for every atom.
```
def printStero(mol):
    info = {}
    _ = AllChem.GetMorganFingerprint(mol, 0, bitInfo=info)
    for element in info:
        for atom_idx, radius_at in info[element]:
            atom = mol.GetAtomWithIdx(atom_idx)
            stereo = atom.GetChiralTag()
            print(stereo)
            
printStereo(SMILES)
```

I was expecting to have some CHI_TETRAHEDRAL_CWs and CHI_TETRAHEDRAL_CCWs since there are "@" and "@@" informations.
Is it about a bug of RDKit since the version is a bit old or is there any other explanation? 

Update : 

I tried with the latest version as well, I still have the same issue. Can someone explain me why? I don't think that it is normal..

Thanks

**Comment by gokhantahil** [comment]
To see exactly the consequences of this behaviour, kekulizing the following two Isomeric SMILES ends up with a SMILES without the tetrahedral chiral information:

`C1=C[C@H]2[C@@H]3[C@H]([C@@H]1[C@H]4[C@@H]2[C@@H]5[C@H]4C6C=C[C@H]7C5[C@H]7C=C6)C=C[C@@H]8[C@H]3[C@@H]9[C@H]8C1C=C[C@H]2C9[C@@H]2C=C1`
![image](https://github.com/rdkit/rdkit/assets/41631875/426840d0-d021-49a6-a97b-dbcbcc50fa94)

`C1=C[C@@H]2[C@H]3[C@H]([C@@H]1[C@H]4[C@@H]2[C@@H]5[C@H]4C6C=C[C@H]7C5[C@H]7C=C6)C=C[C@@H]8[C@H]3[C@@H]9[C@H]8C1C=C[C@H]2C9[C@@H]2C=C1`
![image](https://github.com/rdkit/rdkit/assets/41631875/1c99576b-dff7-4326-88ca-3d276668adff)

`C1=CC2C3C(C1C4C2C5C4C6C=CC7C5C7C=C6)C=CC8C3C9C8C1C=CC2C9C2C=C1`
![image](https://github.com/rdkit/rdkit/assets/41631875/e5db56f1-9eda-45dc-bc9d-08632eda8e48)

**Reply by greglandrum** [reply]
Hi @gokhantahil. This is the result of a bug in the legacy (but still default) stereo perception code.
If you use the new stereo perception, the chiral centers in this molecule are preserved:
```
Chem.SetUseLegacyStereoPerception(False)
m = Chem.MolFromSmiles('C1=C[C@H]2[C@@H]3[C@H]([C@@H]1[C@H]4[C@@H]2[C@@H]5[C@H]4C6C=C[C@H]7C5[C@H]7C=C6)C=C[C@@H]8[C@H]3[C@@H]9[C@H]8C1C=C[C@H]2C9[C@@H]2C=C1')
Chem.MolToSmiles(m)
```
The output from this is:
```
C1=C[C@@H]2C3[C@H]4[C@@H](C1C=C[C@@H]32)[C@H]1[C@@H]2C=C[C@@H]([C@@H]3[C@H]2C=C[C@H]2[C@@H]5C6C=C[C@H]7C([C@@H]5[C@H]23)[C@@H]7C=C6)[C@@H]41
```

The plan is that the new stereo perception code will be the default in the next major RDKit release (2023.09.1).

**Comment by gokhantahil** [comment]
Thanks a lot, it works without any problem with these molecules! @greglandrum

**Comment by gokhantahil** [comment]
Hello again,

After making some tests with the RDKit version 2023.03.1, I've seen that for the some SMILES, the double bond configuration informations (cis-trans or E-Z or "/"-"\\") of some molecules has been lost after using the 
`Chem.SetUseLegacyStereoPerception(False)`

Here are the scripts that I've tested with 2 SMILES that I had this issue :

```
from rdkit import Chem
Chem.SetUseLegacyStereoPerception(False)
def printKekulize(mol):
    molObject = Chem.MolFromSmiles(mol)
    print(Chem.MolToSmiles(molObject, kekuleSmiles=True))

SMILES = ['[H]/N=c/1\\n(c2c(s1)C[C@H](CC2)[NH3+])C3CCC4(CC3)C[C@@H]4C(=O)OCC',
          '[H]/N=c\\1/n(c2c(s1)C[C@H](CC2)[NH3+])C3CCC4(CC3)C[C@@H]4C(=O)OCC']

for smiles in SMILES:
    printKekulize(smiles)
```
returns the following SMILES:

```
[H]N=C1SC2=C(CC[C@H]([NH3+])C2)N1C1CCC2(CC1)C[C@@H]2C(=O)OCC
[H]N=C1SC2=C(CC[C@H]([NH3+])C2)N1C1CCC2(CC1)C[C@@H]2C(=O)OCC
```

However, without stere perception, it is not the case:

```
from rdkit import Chem
def printKekulize(mol):
    molObject = Chem.MolFromSmiles(mol)
    print(Chem.MolToSmiles(molObject, kekuleSmiles=True))

SMILES = ['[H]/N=c/1\\n(c2c(s1)C[C@H](CC2)[NH3+])C3CCC4(CC3)C[C@@H]4C(=O)OCC',
          '[H]/N=c\\1/n(c2c(s1)C[C@H](CC2)[NH3+])C3CCC4(CC3)C[C@@H]4C(=O)OCC']

for smiles in SMILES:
    printKekulize(smiles)
```

returns the following SMILES:

```
[H]/N=C1/SC2=C(CC[C@H]([NH3+])C2)N1C1CCC2(CC1)C[C@@H]2C(=O)OCC
[H]/N=C1\SC2=C(CC[C@H]([NH3+])C2)N1C1CCC2(CC1)C[C@@H]2C(=O)OCC
```

I believe that it is not normal, if it is the case, can you inform me?

Thanks!

**Reply by greglandrum** [reply, marked as answer]
I believe that this is a bug in the new stereo perception code for the case of the H attached to an N. Here's a simpler illustration:
```
In [22]: Chem.SetUseLegacyStereoPerception(True)

In [23]: Chem.CanonSmiles('[H]/N=C/F')
Out[23]: '[H]/N=C/F'

In [24]: Chem.CanonSmiles('C/N=C/F')
Out[24]: 'C/N=C/F'

In [25]: Chem.SetUseLegacyStereoPerception(False)

In [26]: Chem.CanonSmiles('[H]/N=C/F')
Out[26]: '[H]N=CF'

In [27]: Chem.CanonSmiles('C/N=C/F')
Out[27]: 'C/N=C/F'
```

**Reply by greglandrum** [reply]
As an aside: please start a new discussion for a new topic in the future... that will make it easier to find things.

**Comment by gokhantahil** [comment]
Thanks a lot!
